### ISSUE TITLE
Feature request: Draft custom sets with a new tournament type, Booster Draft Swiss|Elimination (Custom)

### ISSUE THREAD

**Issue by T-and-Em** [magefree/mage#8079]
When choosing the set(s) for drafting it would be great if we could choose sets of our own creation. A custom set would be a simple text file containing a list of cards and their rarity in the set. Perhaps something like this:

```
NAME: Ice Age Remastered
DESC: Custom set using cards from Ice Age, Alliances, Coldsnap, Future Sight, Modern Horizons and Kaldheim
OPTS:
include_basic_lands=false
lands=1 # if include_basic_lands is false and this is > 0, choose a land from the card list below
commons=10
uncommons=3
rare=1
mythic_ratio=7.5

CARDS:
# rarity [set:num] name
C [ME2:245] Snow-Covered Forest
C [ICE:100] Snow Devil
U [CSP:123] Sound the Call
U [MH1:194] Abominable Treefolk
...

```
When generating booster packs the rarity of the card specified in the custom set file would be used, not the rarity the card has printed on it. For example, in the custom set above the card _Sound the Call_ has been shifted up from a common to an uncommon.

Alternatively, instead of creating a new tournament type, instead when choosing packs in the existing Swiss or Elimination draft types add an option to the top or bottom of the pack list to allow loading of the custom set.

Bonus effect of implementing this feature: Whenever an official 'remastered' set is completely spoiled the XMage community could implement it as a custom set while waiting for a new XMage release to include it.

**Comment by JayDi85** [comment]
There are custom cube from the deck (you can use any cards for your tourney and it will be put with boosters). But it uses different booster logic (puts cards to booster at random order, so rarity ratio can be different).

![shot_210731_095828](https://user-images.githubusercontent.com/8344157/127730513-43c4ffec-0a7a-4a92-b3e0-81ff82ce531d.png)

**Comment by T-and-Em** [comment]
> 
> 
> There are custom cube from the deck (you can use any cards for your tourney and it will be put with boosters). But it uses different booster logic (puts cards to booster at random order, so rarity ratio can be different).
> 
> ![shot_210731_095828](https://user-images.githubusercontent.com/8344157/127730513-43c4ffec-0a7a-4a92-b3e0-81ff82ce531d.png)

I believe the booster logic for sets is different enough from cubes for this to justify it's own request.

**Comment by JayDi85** [comment]
> I believe the booster logic for sets is different enough from cubes for this to justify it's own request.

You can create any request or question. Current request is overdeveloped and duplicates another features. 

Problems:
- hard to implement and support:
  - need special parser for new file format;
  - need support in deck editor to load and save that file -- without gui editor no one will use it
  - need custom set support in different game modes;
  - gui doesn't support custom rarity for cards, so it need improves too (and I don't know how many). Same for server side I think.
- duplicated features from "Cube from Deck" mode (95% are same).

If you remove custom rarity then it will be... Cube from Deck. Already workable, with deck editor and game modes support.

**Comment by T-and-Em** [comment]
You're right - If people can't use a GUI of some kind to create the custom sets it would likely not get used...

> 
>     * duplicated features from "Cube from Deck" mode (95% are same).
> 

It's that 5% difference that matters. Would it be too much to add a selector to the New Tourney dialog to choose between cube collation and set collation? Something like this:
![image](https://user-images.githubusercontent.com/76133892/127738583-73910de2-218f-4f13-bca5-61cdd40d45f0.png)

This would control how the server generates the packs. Ideally choosing set collation would pop up another dialog to customize the ratio of rarities, but if that's too much just use the standard 1 land, 10 commons, 3 uncommons, 1 rare.

**Comment by JayDi85** [comment]
Collation methods for cube can be implemented:
- Random 15 (current);
- Standard rarity slots (xmage will split a cube cards by rarity and fill each booster due rarity slots);

**Comment by T-and-Em** [comment]
> 
> 
> Collation methods for cube can be implemented:
> 
>     * Random 15 (current);
> 
>     * Standard rarity slots (xmage will split a cube cards by rarity and fill each booster due rarity slots);

That would be fantastic! Thank you!!

**Comment by theelk801** [comment]
I have some unfinished code for a color-based collation method that I've been meaning to finish
